### DISCUSSION TITLE
CLI: Get html content of the saved page

### DISCUSSION THREAD

**Discussion by mayursarojdev** [gildas-lormeau/SingleFile#1087] [Q&A]
Hello, is there a way to get the html content of saved singleFile html without having to read the .html file. Going through the cli options `--dump-content`  sounded like it logs the html content as string in console but trying with the option `--dump-content=true` with `--back-end=puppeteer` it did not return the content. 
Any pointers on this would be appreciated.

Thank you

**Comment by Nirmal-Kotiyal** [comment]
> Hello, is there a way to get the html content of saved singleFile html without having to read the .html file. Going through the cli options `--dump-content` sounded like it logs the html content as string in console but trying with the option `--dump-content=true` with `--back-end=puppeteer` it did not return the content. Any pointers on this would be appreciated.
> 
> Thank you

Yes, it would be very helpful if there was a way we can directly get the html content of the url opened through single-file-cli

**Comment by gildas-lormeau** [comment]
Can you provide an example which fails? The following command works fine for me. Puppeteer is the default back-end.
```
% ./single-file https://www.example.com --dump-content
<!DOCTYPE html> <html><!--
 Page saved with SingleFile 
 url: https://www.example.com 
 ...
```

**Reply by mayursarojdev** [reply]
Thank you for response, I used `single-file https://www.example.com example.html --dump-content=true`
From your response i realised if iam not wrong `--dump-content` cannot be used in conjunction with saving .html. The usecase i was looking for was where singleFile saves .html but also returns the html string so the .html wouldnt needed to be read.

**Reply by mayursarojdev** [reply]
> Can you provide an example which fails? The following command works fine for me. Puppeteer is the default back-end.
> 
> ```
> % ./single-file https://www.example.com --dump-content
> <!DOCTYPE html> <html><!--
>  Page saved with SingleFile 
>  url: https://www.example.com 
>  ...
> ```

Hello, would really appreciate if you could confirm whether it is possible to save singleFile .html and also return the html contents string.

**Reply by gildas-lormeau** [reply]
I confirm that's it's not possible currently. It requires some changes in the code.

**Reply by mayursarojdev** [reply]
Thank you for your response.
